src/server.h

```diff
@@ -2027,6 +2027,8 @@ struct valkeyServer {
     unsigned long long cluster_link_msg_queue_limit_bytes;  /* Memory usage limit on individual link msg queue */
     int cluster_drop_packet_filter; /* Debug config that allows tactically
                                    * dropping packets of a specific type */
+    int cluster_close_link_on_packet_drop; /* Debug config that goes along with cluster_drop_packet_filter.
+                                              When set, the link is closed on packet drop. */
```

Comment:
Is there any alternative you thought of for testing without introducing this particular flag?

Reply:
In order to test this, the target node has to close the physical connection and not just simply drop the packet. Simply dropping the packet is not a valid scenario in production as we use TCP and we expect reliable transmission.

The only other approach that I can think of is to use iptables to drop packets while the connection is setup. I couldn't find examples in tcl tests that use iptables. Do you have a better approach in mind?

Reply:
In the cluster tests for failover, we use SIGPAUSE to stop node. (There's some TCL helper function for it, like `proc pause_process` I think.) Then, it doesn't respond to any network traffic. SIGCONT is used to wake it up again.

Reply:
SIGPAUSE and SIGCONT may not reproduce the scenario that this PR is fixing. We want the connect to be established successfully so that the sender sends the MEET message. The connection has to be torn down before the MEET is received by the receiver. This makes it tricky.

Reply:
OK